### ISSUE TITLE
[1.2.0] - Unable to publish from Azure DevOps - Error: Specified webPartData is not a valid JSON string.

### ISSUE THREAD

**Issue by garrytrinder** [ValoIntranet/doctor#16]
### Description

Error publishing page from Azure DevOps Ubuntu agent

### Steps to reproduce

Given the folder structure

- src
	- home.md

and

<details>
<summary>home.md</summary>

<pre>
---
title: Home
slug: home.aspx
draft: false

menu:
  QuickLaunch:
    id: Home
    weight: 1
---

# Doctor

Welcome to the static page created by `Doctor`!
</pre>
</details>

Execute

```sh
doctor publish --url https://contoso.sharepoint.com/teams/site --auth password --username <username> --password <password>
````

### Expected result

Page is created successfully

### Actual result

`Error: Specified webPartData is not a valid JSON string.`

<details>
<summary>Log</summary>
<pre>
2020-12-18T16:48:45.9065154Z ##[section]Starting: Job
2020-12-18T16:48:46.0734569Z ##[section]Starting: Initialize job
2020-12-18T16:48:46.0735716Z Agent name: 'Hosted Agent'
2020-12-18T16:48:46.0736052Z Agent machine name: 'fv-az50-373'
2020-12-18T16:48:46.0736254Z Current agent version: '2.179.0'
2020-12-18T16:48:46.0774719Z ##[group]Operating System
2020-12-18T16:48:46.0774916Z Ubuntu
2020-12-18T16:48:46.0775035Z 18.04.5
2020-12-18T16:48:46.0775147Z LTS
2020-12-18T16:48:46.0775286Z ##[endgroup]
2020-12-18T16:48:46.0775435Z ##[group]Virtual Environment
2020-12-18T16:48:46.0775608Z Environment: ubuntu-18.04
2020-12-18T16:48:46.0775783Z Version: 20201210.0
2020-12-18T16:48:46.0776054Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu18/20201210.0/images/linux/Ubuntu1804-README.md
2020-12-18T16:48:46.0776315Z ##[endgroup]
2020-12-18T16:48:46.0777257Z Current image version: '20201210.0'
2020-12-18T16:48:46.0781787Z Agent running as: 'vsts'
2020-12-18T16:48:46.0824649Z Prepare build directory.
2020-12-18T16:48:46.1068464Z Set build variables.
2020-12-18T16:48:46.1099456Z Download all required tasks.
2020-12-18T16:48:46.1209602Z Downloading task: NodeTool (0.179.1)
2020-12-18T16:48:47.9785709Z Downloading task: CmdLine (2.178.0)
2020-12-18T16:48:48.2037553Z Checking job knob settings.
2020-12-18T16:48:48.2045794Z    Knob: AgentToolsDirectory = /opt/hostedtoolcache Source: ${AGENT_TOOLSDIRECTORY} 
2020-12-18T16:48:48.2046814Z    Knob: AgentPerflog = /home/vsts/perflog Source: ${VSTS_AGENT_PERFLOG} 
2020-12-18T16:48:48.2047465Z Finished checking job knob settings.
2020-12-18T16:48:48.2534851Z Start tracking orphan processes.
2020-12-18T16:48:48.2733077Z ##[section]Finishing: Initialize job
2020-12-18T16:48:48.3070580Z ##[section]Starting: Checkout ValoDoctor@azure-pipelines to s
2020-12-18T16:48:48.3289323Z ==============================================================================
2020-12-18T16:48:48.3290081Z Task         : Get sources
2020-12-18T16:48:48.3290635Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-12-18T16:48:48.3290924Z Version      : 1.0.0
2020-12-18T16:48:48.3291486Z Author       : Microsoft
2020-12-18T16:48:48.3292150Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2020-12-18T16:48:48.3292453Z ==============================================================================
2020-12-18T16:48:48.7379416Z Syncing repository: ValoDoctor (Git)
2020-12-18T16:48:49.4856834Z ##[command]git version
2020-12-18T16:48:49.4862826Z git version 2.29.2
2020-12-18T16:48:49.4864584Z ##[command]git lfs version
2020-12-18T16:48:49.4870320Z git-lfs/2.12.1 (GitHub; linux amd64; go 1.13.4)
2020-12-18T16:48:49.4872804Z ##[command]git init "/home/vsts/work/1/s"
2020-12-18T16:48:49.4885854Z Initialized empty Git repository in /home/vsts/work/1/s/.git/
2020-12-18T16:48:49.4887440Z ##[command]git remote add origin https://cpsglobal-dev@dev.azure.com/cpsglobal-dev/ValoDoctor/_git/ValoDoctor
2020-12-18T16:48:49.4897291Z ##[command]git config gc.auto 0
2020-12-18T16:48:49.4906368Z ##[command]git config --get-all http.https://cpsglobal-dev@dev.azure.com/cpsglobal-dev/ValoDoctor/_git/ValoDoctor.extraheader
2020-12-18T16:48:49.4914016Z ##[command]git config --get-all http.proxy
2020-12-18T16:48:49.4947591Z ##[command]git config http.version HTTP/1.1
2020-12-18T16:48:49.4957906Z ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --force --tags --prune --progress --no-recurse-submodules origin
2020-12-18T16:48:49.4963863Z remote: Azure Repos        
2020-12-18T16:48:49.4964158Z remote: 
2020-12-18T16:48:49.4964396Z remote: Found 48 objects to send. (86 ms)        
2020-12-18T16:48:49.4964923Z From https://dev.azure.com/cpsglobal-dev/ValoDoctor/_git/ValoDoctor
2020-12-18T16:48:49.4965912Z  * [new branch]      azure-pipelines -> origin/azure-pipelines
2020-12-18T16:48:49.4966537Z  * [new branch]      main            -> origin/main
2020-12-18T16:48:49.5434413Z ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --force --tags --prune --progress --no-recurse-submodules origin +87d22f51fefe0f519511faabfe302d96713aabd3:refs/remotes/origin/87d22f51fefe0f519511faabfe302d96713aabd3
2020-12-18T16:48:49.5680499Z From https://dev.azure.com/cpsglobal-dev/ValoDoctor/_git/ValoDoctor
2020-12-18T16:48:49.5682278Z  * [new ref]         87d22f51fefe0f519511faabfe302d96713aabd3 -> origin/87d22f51fefe0f519511faabfe302d96713aabd3
2020-12-18T16:48:49.6076452Z ##[command]git checkout --progress --force 87d22f51fefe0f519511faabfe302d96713aabd3
2020-12-18T16:48:49.6080351Z Note: switching to '87d22f51fefe0f519511faabfe302d96713aabd3'.
2020-12-18T16:48:49.6080546Z 
2020-12-18T16:48:49.6080984Z You are in 'detached HEAD' state. You can look around, make experimental
2020-12-18T16:48:49.6081313Z changes and commit them, and you can discard any commits you make in this
2020-12-18T16:48:49.6081728Z state without impacting any branches by switching back to a branch.
2020-12-18T16:48:49.6081916Z 
2020-12-18T16:48:49.6082116Z If you want to create a new branch to retain commits you create, you may
2020-12-18T16:48:49.6082617Z do so (now or later) by using -c with the switch command. Example:
2020-12-18T16:48:49.6082875Z 
2020-12-18T16:48:49.6083195Z   git switch -c <new-branch-name>
2020-12-18T16:48:49.6083329Z 
2020-12-18T16:48:49.6083625Z Or undo this operation with:
2020-12-18T16:48:49.6083761Z 
2020-12-18T16:48:49.6084050Z   git switch -
2020-12-18T16:48:49.6084158Z 
2020-12-18T16:48:49.6084365Z Turn off this advice by setting config variable advice.detachedHead to false
2020-12-18T16:48:49.6084551Z 
2020-12-18T16:48:49.6084928Z HEAD is now at 87d22f5 Update azure-pipelines.yml for Azure Pipelines
2020-12-18T16:48:49.6135914Z ##[section]Finishing: Checkout ValoDoctor@azure-pipelines to s
2020-12-18T16:48:49.6160659Z ##[section]Starting: Install Node@14
2020-12-18T16:48:49.6167016Z ==============================================================================
2020-12-18T16:48:49.6167307Z Task         : Node.js tool installer
2020-12-18T16:48:49.6167637Z Description  : Finds or downloads and caches the specified version spec of Node.js and adds it to the PATH
2020-12-18T16:48:49.6167936Z Version      : 0.179.1
2020-12-18T16:48:49.6168143Z Author       : Microsoft Corporation
2020-12-18T16:48:49.6168571Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/tool/node-js
2020-12-18T16:48:49.6168914Z ==============================================================================
2020-12-18T16:48:50.1479424Z Found tool in cache: node 14.15.1 x64
2020-12-18T16:48:50.1482494Z Prepending PATH environment variable with directory: /opt/hostedtoolcache/node/14.15.1/x64/bin
2020-12-18T16:48:50.1935140Z ##[section]Finishing: Install Node@14
2020-12-18T16:48:50.1959783Z ##[section]Starting: Install Valo Intranet Doctor
2020-12-18T16:48:50.1965988Z ==============================================================================
2020-12-18T16:48:50.1966262Z Task         : Command line
2020-12-18T16:48:50.1966561Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2020-12-18T16:48:50.1966843Z Version      : 2.178.0
2020-12-18T16:48:50.1967045Z Author       : Microsoft Corporation
2020-12-18T16:48:50.1967347Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2020-12-18T16:48:50.1967677Z ==============================================================================
2020-12-18T16:48:50.3406826Z Generating script.
2020-12-18T16:48:50.3415848Z Script contents:
2020-12-18T16:48:50.3416527Z npm i -g @valo/doctor@latest
2020-12-18T16:48:50.3416845Z ========================== Starting Command Output ===========================
2020-12-18T16:48:50.3441943Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/04157bd4-25c8-4943-ab4e-d57dd21456cc.sh
2020-12-18T16:49:00.1870694Z npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
2020-12-18T16:49:00.1873545Z npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
2020-12-18T16:49:01.1472228Z npm WARN deprecated har-validator@5.1.5: this library is no longer supported
2020-12-18T16:49:14.4145503Z /opt/hostedtoolcache/node/14.15.1/x64/bin/doctor -> /opt/hostedtoolcache/node/14.15.1/x64/lib/node_modules/@valo/doctor/bin/doctor
2020-12-18T16:49:14.4154917Z /opt/hostedtoolcache/node/14.15.1/x64/bin/localm365 -> /opt/hostedtoolcache/node/14.15.1/x64/lib/node_modules/@valo/doctor/bin/localm365
2020-12-18T16:49:14.6864287Z npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/@valo/doctor/node_modules/@pnp/cli-microsoft365/node_modules/fsevents):
2020-12-18T16:49:14.6867740Z npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
2020-12-18T16:49:14.6868378Z 
2020-12-18T16:49:14.6896722Z + @valo/doctor@1.2.0
2020-12-18T16:49:14.6897057Z added 648 packages from 879 contributors in 19.284s
2020-12-18T16:49:14.7654844Z ##[section]Finishing: Install Valo Intranet Doctor
2020-12-18T16:49:14.7676154Z ##[section]Starting: Publish Pages
2020-12-18T16:49:14.7684160Z ==============================================================================
2020-12-18T16:49:14.7684437Z Task         : Command line
2020-12-18T16:49:14.7684721Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2020-12-18T16:49:14.7684999Z Version      : 2.178.0
2020-12-18T16:49:14.7685200Z Author       : Microsoft Corporation
2020-12-18T16:49:14.7685493Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2020-12-18T16:49:14.7685835Z ==============================================================================
2020-12-18T16:49:14.9030048Z Generating script.
2020-12-18T16:49:14.9040658Z Script contents:
2020-12-18T16:49:14.9042204Z doctor publish --url https://cpsglobaldev.sharepoint.com/teams/ValoDoctor --auth password --username *** --password ***
2020-12-18T16:49:14.9043013Z ========================== Starting Command Output ===========================
2020-12-18T16:49:14.9066066Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/7f2339d5-21a0-4be5-a6be-747cfbdeaa76.sh
2020-12-18T16:49:17.4514884Z 
2020-12-18T16:49:17.4523204Z  START:  publish job
2020-12-18T16:49:17.4524505Z 
2020-12-18T16:49:21.3771955Z [16:49:21] Authenticate to M365 with password [started]
2020-12-18T16:49:22.4979539Z [16:49:22] Authenticate to M365 with password [completed]
2020-12-18T16:49:22.4983627Z [16:49:22] Fetch all markdown files [started]
2020-12-18T16:49:22.5104853Z [16:49:22] Fetch all markdown files [completed]
2020-12-18T16:49:22.5113973Z [16:49:22] Process markdown files [started]
2020-12-18T16:49:22.5145691Z [16:49:22] â†’ Started processing: home.md
2020-12-18T16:49:22.6102067Z [16:49:22] â†’ Creating or updating the page in SharePoint for home.md
2020-12-18T16:49:27.3432630Z [16:49:27] Process markdown files [failed]
2020-12-18T16:49:27.3439204Z [16:49:27] â†’ Command failed: localm365 spo page control set --webUrl "https://cpsglobaldev.sharepoint.com/teams/ValoDoctor" --name "home.aspx" --id "5af55e94-16b1-4d20-aa8e-972af43a482f" --webPartData '{"title":"doctor-placeholder","serverProcessedContent": {"searchablePlainTexts": {"code": "
2020-12-18T16:49:27.3461009Z \n# Doctor
2020-12-18T16:49:27.3461996Z \n
2020-12-18T16:49:27.3464179Z \nWelcome to the static page created by `Doctor`!"}},"dataVersion": "2.0","properties": {"displayPreview": true,"lineWrapping": true,"miniMap": {"enabled": false},"previewState": "Show","theme": "Monokai"}}'
2020-12-18T16:49:27.3466189Z Error: Specified webPartData is not a valid JSON string. Input: {"title":"doctor-placeholder","serverProcessedContent": {"searchablePlainTexts": {"code": "
2020-12-18T16:49:27.3471282Z \n# Doctor
2020-12-18T16:49:27.3472274Z \n
2020-12-18T16:49:27.3473574Z \nWelcome to the static page created by `Doctor`!"}},"dataVersion": "2.0","properties": {"displayPreview": true,"lineWrapping": true,"miniMap": {"enabled": false},"previewState": "Show","theme": "Monokai"}}. Error: SyntaxError: Unexpected token 
2020-12-18T16:49:27.3475160Z  in JSON at position 91
2020-12-18T16:49:27.3476004Z 
2020-12-18T16:49:27.3477677Z Error: Command failed: localm365 spo page control set --webUrl "https://cpsglobaldev.sharepoint.com/teams/ValoDoctor" --name "home.aspx" --id "5af55e94-16b1-4d20-aa8e-972af43a482f" --webPartData '{"title":"doctor-placeholder","serverProcessedContent": {"searchablePlainTexts": {"code": "
2020-12-18T16:49:27.3478573Z \n# Doctor
2020-12-18T16:49:27.3478870Z \n
2020-12-18T16:49:27.3479898Z \nWelcome to the static page created by `Doctor`!"}},"dataVersion": "2.0","properties": {"displayPreview": true,"lineWrapping": true,"miniMap": {"enabled": false},"previewState": "Show","theme": "Monokai"}}'
2020-12-18T16:49:27.3481141Z Error: Specified webPartData is not a valid JSON string. Input: {"title":"doctor-placeholder","serverProcessedContent": {"searchablePlainTexts": {"code": "
2020-12-18T16:49:27.3481762Z \n# Doctor
2020-12-18T16:49:27.3482073Z \n
2020-12-18T16:49:27.3482931Z \nWelcome to the static page created by `Doctor`!"}},"dataVersion": "2.0","properties": {"displayPreview": true,"lineWrapping": true,"miniMap": {"enabled": false},"previewState": "Show","theme": "Monokai"}}. Error: SyntaxError: Unexpected token 
2020-12-18T16:49:27.3483714Z  in JSON at position 91
2020-12-18T16:49:27.3484001Z 
2020-12-18T16:49:27.3484341Z     at ChildProcess.exithandler (child_process.js:308:12)
2020-12-18T16:49:27.3484774Z     at ChildProcess.emit (events.js:315:20)
2020-12-18T16:49:27.3485236Z     at ChildProcess.EventEmitter.emit (domain.js:486:12)
2020-12-18T16:49:27.3485666Z     at maybeClose (internal/child_process.js:1048:16)
2020-12-18T16:49:27.3486099Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5) {
2020-12-18T16:49:27.3486654Z   killed: false,
2020-12-18T16:49:27.3486936Z   code: 1,
2020-12-18T16:49:27.3487335Z   signal: null,
2020-12-18T16:49:27.3489340Z   cmd: 'localm365 spo page control set --webUrl "https://cpsglobaldev.sharepoint.com/teams/ValoDoctor" --name "home.aspx" --id "5af55e94-16b1-4d20-aa8e-972af43a482f" --webPartData \'{"title":"doctor-placeholder","serverProcessedContent": {"searchablePlainTexts": {"code": "\r\\n# Doctor\r\\n\r\\nWelcome to the static page created by `Doctor`!"}},"dataVersion": "2.0","properties": {"displayPreview": true,"lineWrapping": true,"miniMap": {"enabled": false},"previewState": "Show","theme": "Monokai"}}\'',
2020-12-18T16:49:27.3490809Z   context: [Object: null prototype] {
2020-12-18T16:49:27.3491399Z     files: [ '/home/vsts/work/1/s/src/home.md' ]
2020-12-18T16:49:27.3491815Z   }
2020-12-18T16:49:27.3492176Z }
2020-12-18T16:49:27.3492393Z 
2020-12-18T16:49:27.3492703Z  Publishing stats 
2020-12-18T16:49:27.3492988Z  Pages: 0
2020-12-18T16:49:27.3493257Z  Images: 0
2020-12-18T16:49:27.3493527Z 
2020-12-18T16:49:27.3493814Z  EXECUTION TIME:  9sec.
2020-12-18T16:49:27.3494140Z  COMPLETED:  publish job
2020-12-18T16:49:27.3494446Z 
2020-12-18T16:49:27.4202563Z ##[section]Finishing: Publish Pages
2020-12-18T16:49:27.4230622Z ##[section]Starting: Checkout ValoDoctor@azure-pipelines to s
2020-12-18T16:49:27.4237286Z ==============================================================================
2020-12-18T16:49:27.4237567Z Task         : Get sources
2020-12-18T16:49:27.4237862Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2020-12-18T16:49:27.4238414Z Version      : 1.0.0
2020-12-18T16:49:27.4238593Z Author       : Microsoft
2020-12-18T16:49:27.4238874Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
2020-12-18T16:49:27.4239203Z ==============================================================================
2020-12-18T16:49:27.7388584Z Cleaning any cached credential from repository: ValoDoctor (Git)
2020-12-18T16:49:27.7435287Z ##[section]Finishing: Checkout ValoDoctor@azure-pipelines to s
2020-12-18T16:49:27.7489012Z ##[section]Starting: Finalize Job
2020-12-18T16:49:27.7517999Z Cleaning up task key
2020-12-18T16:49:27.7519116Z Start cleaning up orphan processes.
2020-12-18T16:49:27.7679884Z ##[section]Finishing: Finalize Job
2020-12-18T16:49:27.7716826Z ##[section]Finishing: Job
</pre>
</details>

### Environment

Azure DevOps
Ubuntu@18.04.5
bash
Node@14
@valo/doctor@latest (1.2.0)

**Comment by garrytrinder** [comment]
The webPartData being passed in to the CLI command is 

```json
{
    "title": "doctor-placeholder",
    "serverProcessedContent": {
        "searchablePlainTexts": {
            "code": "\r\\nHello\r\\n"
        }
    },
    "dataVersion": "2.0",
    "properties": {
        "displayPreview": true,
        "lineWrapping": true,
        "miniMap": {
            "enabled": false
        },
        "previewState": "Show",
        "theme": "Monokai"
    }
}
```

but in the error, it is returned as...

```json
{
    "title": "doctor-placeholder",
    "serverProcessedContent": {
        "searchablePlainTexts": {
            "code\n"
        }
    },
    "dataVersion": "2.0",
    "properties": {
        "displayPreview": true,
        "lineWrapping": true,
        "miniMap": {
            "enabled": false
        },
        "previewState": "Show",
        "theme": "Monokai"
    }
}
```

**Comment by garrytrinder** [comment]
Confirmed that this is fixed in the latest beta.

Thanks @estruyf great work 🔥🚀

**Comment by estruyf** [comment]
Thanks @garrytrinder for testing this! Now available in version `1.2.1`.
